scripts/ImportExportTest.sh

```diff
@@ -0,0 +1,270 @@
+                    return 0
+                else
+                    for output in "${outputs[@]}"
+                    do
+                        for obtained_contract in $(jq '.contracts | keys | .[]' obtained.json  2> /dev/null)
+                        do
+                            jq --raw-output ".contracts.${obtained_contract}.\"${output}\"" obtained.json > "obtained.${output}"
+                            set +e
+                            diff_files "expected.${output}" "obtained.${output}"
+                            DIFF=$?
+                            set -euo pipefail
+                            if [[ ${DIFF} != 0 ]]
+                            then
+                                _TESTED=
+                                FAILED=$((FAILED + 1))
+                                return 0
+                            fi
+                        done
+                    done
+
+                    # direct export via --asm-json, if imported with --import-asm-json.
+                    if ! $SOLC --asm-json --import-asm-json expected.asm | tail -n+4 > obtained_direct_import_export.json 2> obtained_direct_import_export.error
+                    then
+                        printError
+                        printError "$nth_input_file"
+                        cat obtained_direct_import_export.error >&2
+                        FAILED=$((FAILED + 1))
+                        return 0
+                    else
+                        for obtained_contract in $(jq '.contracts | keys | .[]' obtained_direct_import_export.json 2> /dev/null)
+                        do
+                            jq --raw-output ".contracts.${obtained_contract}.\"asm\"" obtained_direct_import_export.json > obtained_direct_import_export.asm
+                            set +e
+                            diff_files "expected.asm" "obtained_direct_import_export.asm"
+                            DIFF=$?
+                            set -euo pipefail
+                            if [[ ${DIFF} != 0 ]]
+                            then
+                                _TESTED=
+                                FAILED=$((FAILED + 1))
+                                return 0
+                            fi
+                        done
+                    fi
+                fi
+            fi
+        done
+    fi
+    if [ -n "${_TESTED}" ]
+    then
+        TESTED=$((TESTED + 1))
+    fi
+}
+
+# function tests whether exporting and importing again leaves the JSON ast unchanged
+# Results are recorded by adding to FAILED or UNCOMPILABLE.
+# Also, in case of a mismatch a diff and the respective ASTs are printed
+# Expected parameters:
+# $1 name of the file to be exported and imported
+# $2 any files needed to do so that might be in parent directories
+function testImportExportEquivalence {
+    local nth_input_file="$1"
+    IFS=" " read -r -a all_input_files <<< "$2"
+
+    if [ -z ${all_input_files+x} ]
+    then
+        all_input_files=( "" )
```

Comment:
Isn't this equivalent to a single input file with an empty name? Didn't you mean this?

```suggestion
        all_input_files=()
```

Reply:
Yeah I also tried this one. Sadly it seem not to work.
```
line 162: all_input_files[@]: unbound variable
```
Looks like that there must be at least one element inside.

Reply:
Oh. But does it work? Because the problem I see is that `solc` gets an empty value as a file name and will try to load that from disk. Or does `( "" )` evaluate to an empty array or something?